src/Utilities/RunTimeManager.cpp

```diff
@@ -129,6 +130,21 @@ bool RunTimeControl<CLOCK>::checkStop(LoopTimer<CLOCK>& loop_timer)
   return need_to_stop;
 }
 
+template<class CLOCK>
+std::string RunTimeControl<CLOCK>::generateProgressMessage(const std::string& driverName,
+                                                           int block,
+                                                           int num_blocks) const
+{
+  std::stringstream log;
+  if (block == 0 || block == num_blocks / 4 || block == num_blocks / 2 || block == (num_blocks * 3) / 4 ||
```

Comment:
I feel better to see the 25th, 50th, 75th and 100th step.
Could you do
```
if (block == 0 || block + 1 == num_blocks / 4 || block + 1 == num_blocks / 2 || block + 1 == (num_blocks * 3) / 4 ||
```
With the 50th step, I can just double the time by 2. With the 51th step, I need additional math.

Reply:
Good suggestion - makes for simple math in many common cases.